sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -160,27 +155,44 @@ export class InteractiveBrowserCredential implements TokenCredential {
         } catch (error) {
           const errorMessage = formatError(
             scopeArray,
-            `${req.query["error"]}. ${req.query["error_description"]}`
+            `${url.searchParams.get("error")}. ${url.searchParams.get("error_description")}`
           );
-          res.status(500).send(errorMessage);
+          res.writeHead(500);
+          res.end(errorMessage);
           logger.getToken.info(errorMessage);
           reject(new Error(errorMessage));
         } finally {
           cleanup();
         }
-      });
+      };
+      const app = http.createServer(requestListener);
 
-      listen = app.listen(this.port, () =>
+      const listen = app.listen(this.port, this.hostname, () =>
         logger.info(`Msal Node Auth Code Sample app listening on port ${this.port}!`)
       );
-      listen.on("connection", (socket) => (socketToDestroy = socket));
+      app.on("connection", (socket) => (socketToDestroy = socket));
+      const server = stoppable(app);
```

Comment:
I was curious about why this is needed - what connections are preventing `app` from stopping gracefully?

If we do want to use this, should we either force-close with a grace of 0 or allow some grace period?

Reply:
Good catch!

Reply:
@maorleger - there's just a long delay on when the server shuts down, afaict. I read through forums on the issue and this package was suggested as a good fix if you want to use the builtin http server.

Reply:
Are there any restrictions on adding third party libraries? I keep forgetting. @bterlson

Reply:
They need to be approved, yes.

@jonathandturner can you say more about the delay you're solving here? Is it because the inbound http connection uses keepalive or something?

Reply:
I'm not sure exactly what causes it but I did notice something similar when we used express. We had to manually delete the express listener for it to shutdown, otherwise it stayed open. For the built-in http sever, this happens even if you manually close it, which is why they made this additional library. The comments on some of the threads are kind of interesting. No one is completely sure why it happens to work this way and generally people just get pointed to this library as the fix.

Reply:
At a quick glance, this dep seems very small/targeted, so it doesn't bother me, though I am curious to learn more about why Node never bothered to fix this natively

Reply:
Thanks for the additional details @jonathandturner. This new dependency seems fine from my perspective.